packages/mask/background/database/post/app.ts

```diff
@@ -0,0 +1,138 @@
+import type { PostRecord as NativePostRecord } from '@masknet/public-api'
+import type {
+    PostRecord,
+    PostReadWriteTransaction,
+    PostReadOnlyTransaction,
+    RecipientDetail,
+    RecipientReason,
+} from './type'
+import {
+    PostIVIdentifier,
+    Identifier,
+    AESJsonWebKey,
+    IdentifierMap,
+    PersonaIdentifier,
+    ProfileIdentifier,
+    ECKeyIdentifier,
+} from '@masknet/shared-base'
+import { nativeAPI } from '../../../shared/native-rpc'
+import { unreachable } from '@dimensiondev/kit'
+
+export async function createPostDB(record: PostRecord, t?: PostReadWriteTransaction) {
+    await nativeAPI?.api.create_post({ post: postInNative(record) as NativePostRecord })
+}
+
+export async function updatePostDB(
+    updateRecord: Partial<PostRecord> & Pick<PostRecord, 'identifier'>,
+    mode: 'append' | 'override',
+    t?: PostReadWriteTransaction,
+): Promise<void> {
+    await nativeAPI?.api.update_post({
+        post: postInNative(updateRecord),
+        options: {
+            mode: mode === 'append' ? 0 : mode === 'override' ? 1 : 0,
+        },
+    })
+    return
+}
+
+export async function queryPostDB(record: PostIVIdentifier, t?: PostReadOnlyTransaction): Promise<PostRecord | null> {
+    const result = await nativeAPI?.api.query_post({ identifier: record.toText() })
+    return result ? postOutNative(result) : null
+}
+
+export async function queryPostPagedDB(
+    linked: PersonaIdentifier,
+    options: {
+        network: string
+        userIds: string[]
+        after?: PostIVIdentifier
+        pageOffset?: number
+    },
+    count: number,
+): Promise<PostRecord[]> {
+    const results = await nativeAPI?.api.query_posts({
+        userIds: options.userIds,
+        network: options.network,
+        encryptBy: linked.toText(),
+        pageOption: options.pageOffset
+            ? {
+                  pageSize: count,
+                  pageOffset: options.pageOffset,
+              }
+            : undefined,
+    })
+    if (!results?.length) return []
+    return results.map((r) => postOutNative(r))
+}
+
+function postInNative(record: Partial<PostRecord> & Pick<PostRecord, 'identifier'>): Partial<NativePostRecord> {
+    return {
+        postBy: record.postBy ? record.postBy.toText() : undefined,
+        identifier: record.identifier.toText(),
+        postCryptoKey: record.postCryptoKey,
+        recipients:
+            record.recipients === 'everyone'
+                ? Object.fromEntries(new Map())
+                : record.recipients
+                ? Array.from(record.recipients).map(
```

Comment:
Recipients is a empty map when`record.recipients === 'everyone'`, and an array when `record.recipients != 'everyone'`.
Please always use map by remove `Array.from(record.recipients)`

Reply:
<img width="438" alt="Screen Shot 2022-02-22 at 3 34 11 PM" src="https://user-images.githubusercontent.com/6863518/155085104-f95f1e3e-88eb-4811-8aaf-1f0888b27a7d.png">
Please use the original format